include/substrait/common/NumberUtils.h

```diff
@@ -0,0 +1,20 @@
+/* SPDX-License-Identifier: Apache-2.0 */
+
+#pragma once
+
+#include <iostream>
+
+namespace io::substrait::common {
+
+class NumberUtils {
+ public:
+  static bool is_number(const std::string& value) {
+    std::string::const_iterator it = value.begin();
+    while (it != value.end() && std::isdigit(*it)) {
```

Comment:
`std::isdigit` can lead to undefined behavior with `char` (`std::string::const_iterator` will give you `char`).  Given that this is likely to be called on untrusted (e.g. yaml / proto plans) input we should avoid it.  Can we cast to unsigned char or come up with some other way to do this?  Maybe use [strtol](https://en.cppreference.com/w/c/string/byte/strtol)?

This will not support decimal.  I'm not sure if that is needed.  If it isn't can we name this `is_integer`?

This will not support negative numbers.  Same as above, if that is not needed let's name this `is_nonnegative_integer`

Reply:
Updated